### PR TITLE
chore: unify public method completion to call on the main thread

### PR DESCRIPTION
**Pull request by duyhungtnn**
To keep iOS SDK consistent with Android SDK
https://github.com/bucketeer-io/android-client-sdk/pull/129
https://github.com/bucketeer-io/android-client-sdk/pull/133

### Changes
- All callback on the the BKTClient's public method will run on the main thread.
- Updated tests

### CODE REVIEW

**Comment by duyhungtnn** [comment]
hi @kakcy , 
please help me to review this changes.
/cc @cre8ivejp

File: Bucketeer/Sources/Public/BKTClient.swift [hunk 1]
```diff
@@ -90,12 +96,12 @@ extension BKTClient {
...
                 client.execute { [weak client] in
                     client?.refreshCache()
-                    client?.fetchEvaluations(timeoutMillis: timeoutMillis, completion: completion)
+                    client?.fetchEvaluations(timeoutMillis: timeoutMillis, completion: initializeCompletion)
                 }
                 BKTClient.default = client
             } catch let error {
                 config.logger?.error(error)
-                completion?(error as? BKTError)
+                initializeCompletion(error as? BKTError ?? BKTError.unknown(message: "unknown", error: error))
```
Comment (cre8ivejp) [inline Bucketeer/Sources/Public/BKTClient.swift]:
```suggestion
                initializeCompletion(error as? BKTError ?? BKTError.unknown(message: "unknown error while returning the initialize completion", error: error))
```

**Comment by cre8ivejp** [comment]
@duyhungtnn 
I have updated the docs for this PR.
https://deploy-preview-68--bucketeer-docs.netlify.app/sdk/client-side/ios

**Comment by duyhungtnn** [comment]
@cre8ivejp thank you , I updated your suggestion.
I also read your new docs updated, I think it good.

**Review by kakcy — APPROVED** [review]
thank you 👍 
I thought it would be okay to comment that the callback is being called on the main thread in the part that calls initialization in the example app, but since it's documented, it's okay to leave it as is.

**Comment by cre8ivejp** [comment]
> thank you 👍 I thought it would be okay to comment that the callback is being called on the main thread in the part that calls initialization in the example app, but since it's documented, it's okay to leave it as is.

I'll add it in another PR. Thanks for suggesting!

**Pull request merged**